src/cuda-poh-verify/poh_verify.cu

```diff
@@ -124,6 +125,7 @@ int poh_verify_many(const uint8_t* hashes,
     CUDA_CHK(cudaMemcpyAsync(cur_ctx->num_hashes_arr, num_hashes_arr, num_hashes_size, cudaMemcpyHostToDevice, stream));
 
     int num_blocks = ROUND_UP_DIV(num_elems, NUM_THREADS_PER_BLOCK);
+    num_blocks += num_blocks == 0 ? 1 : 0;
```

Comment:
Can you just bail if num_elems == 0 earlier?

Reply:
Good question, I was seeing the problem even with what should have been `num_blocks=1`, need to check why that was

Reply:
nm, it's just a problem with num_elems==0